packages/smooth_app/lib/data_models/continuous_scan_model.dart

```diff
@@ -167,9 +167,18 @@ class ContinuousScanModel with ChangeNotifier {
   }
 
   Future<bool> _cachedBarcode(final String barcode) async {
-    final Product? product = await _daoProduct.get(barcode);
+    Product? product = await _daoProduct.get(barcode);
     if (product != null) {
-      _addProduct(product, ScannedProductState.CACHED);
+      try {
+        final FetchedProduct fetchedProduct = await _queryBarcode(barcode);
+        if (fetchedProduct.product != null) {
+          product = fetchedProduct.product;
+        }
+      } catch (e) {
```

Comment:
I think you should specify the kind of expected exception here to only filter the one related to the network.
Something like on `MyNetworkException catch (e)`

Reply:
Yeah changed to SockerException here 
```dart
 try {
        final FetchedProduct fetchedProduct = await _queryBarcode(barcode);
        if (fetchedProduct.product != null) {
          product = fetchedProduct.product;
        }
      } on SocketException {
        // Tried to refresh the product from server but failed maybe due to server error or internet connection issue
        // Then it will show the cached product as it is
      }
   ```